ports/unix/mpthreadport.c

```diff
@@ -321,6 +321,26 @@ void mp_thread_mutex_unlock(mp_thread_mutex_t *mutex) {
     // TODO check return value
 }
 
+#if MICROPY_PY_THREAD_RECURSIVE_MUTEX
+
+void mp_thread_recursive_mutex_init(mp_thread_recursive_mutex_t *mutex) {
```

Comment:
Looks like the initialisation of `thread_mutex` at the top of `mp_thread_init` can use this new function.

Reply:
Oh yeah, I hadn't noticed that mutex! Have refactored.